Formula/l/lilypond.rb

```diff
@@ -39,21 +39,27 @@ class Lilypond < Formula
 
   depends_on "bison" => :build # bison >= 2.4.1 is required
   depends_on "fontforge" => :build
-  depends_on "gettext" => :build
   depends_on "pkg-config" => :build
   depends_on "t1utils" => :build
   depends_on "texinfo" => :build # makeinfo >= 6.1 is required
   depends_on "texlive" => :build
+  depends_on "bdw-gc"
   depends_on "fontconfig"
   depends_on "freetype"
+  depends_on "gettext"
```

Comment:
this should be macos dependency

Reply:
@chenrui333 `gettext` is a build dependency on Linux:

https://lilypond.org/doc/v2.24/Documentation/contributor/requirements-for-compiling-lilypond#other

I’m not sure how to express a build dependency on Linux and a required dependency on macOS. Would this work?

```rb
on_linux do
  depends_on "gettext" => :build
end

on_macos do
  depends_on "gettext"
  depends_on "harfbuzz"
end
```

Reply:
yeah, it would be like that